Gordon360/Controllers/NewsController.cs

```diff
@@ -198,5 +199,26 @@ public ActionResult<StudentNewsViewModel> EditPosting(int newsID, [FromBody] Stu
             var result = _newsService.EditPosting(newsID, studentNewsEdit);
             return Ok(result);
         }
+
+        /// <summary>
+        ///  Approve a news posting in the database
+        /// </summary>
+        /// <param name="newsID">The id of the news item to approve</param>
+        /// <returns>The approved news item</returns>
+        /// <remarks>The news item must not be expired and must be unapproved</remarks>
+        [HttpPut]
+        [Route("approvalStatus")]
+        // only SNAdmin is authorized to use this route
+        // Can't use SYB UPDATE operation, because the author is not authorized to approve the post
+        public ActionResult<StudentNewsViewModel> ApprovePosting([FromBody] int newsID)
+        {
+            if (user_groups.Contains(AuthGroup.SiteAdmin) || user_groups.Contains(AuthGroup.NewsAdmin))
```

Comment:
You updated stateyourbusiness but don't reference it here? You should either have your controller handle the authorization logic or have stateyourbusiness do it (preferably SYB). Don't add the logic to both.

Reply:
![image](https://user-images.githubusercontent.com/78691207/216785783-b6f8cfa1-1370-49b8-aec5-9404f81c3a50.png)
Should we allow the author to approve the post? Because the update operation also includes author, so I was concerned and didn't use SYB here.

Reply:
Ah, that's a good clarification on what I was missing. You're additionally trying to allow student news admins to edit posts in StateYourBusiness.cs. That change is separate from the permissions for approving. Sounds good then!